server/dash/dashboards/ncSumComp.py

```diff
@@ -0,0 +1,407 @@
+    ], style=EQUAL_SPACE_STYLE.update({"width": "97.5vw"})),
+
+    html.Div(html.Br(), style=DIVIDER_STYLE),
+
+    # NC Comparison - Number of Requests per day Overlapping line chart
+    html.Div(dcc.Graph(id="overlay_req_time_line_chart", style={"height": "32vh", "width": "97.5vw"}), style=BORDER_STYLE.update({
+        "height": "32vh", "width": "97.5vw"}))
+])
+
+# CALLBACK FUNCTIONS
+@callback(
+    [Output("nc_dropdown_filter", "options"),
+     Output("nc_dropdown_filter", "value")],
+    Input("nc_dropdown", "value")
+)
+def generate_dynamic_filter(nc_dropdown):
+    """Enables the dashboard to show dynamic filters.
+
+    This function takes the selected neighborhood council (nc) value from the "nc_dropdown" dropdown and output a a list of available request types
+    from that neigbhorhood council.
+
+    Args:
+        nc_dropdown: A string argument automatically detected by Dash callback function when "nc_dropdown" element is selected in the layout.
+
+    Returns: 
+        nc_dropdown_filter: a list of request types available from the selected neigbhorhood council.
+        nc_dropdown_filter: placeholder value for the dropdown when nothing is selected.
+    """
+    # If no neighborhood council is selected, use all data.
+    if not nc_dropdown:
+        df = data_2020
+    else:
+        df = data_2020[data_2020["councilName"] == nc_dropdown]
+
+    req_types = sorted([n for n in set(df["typeName"])])
+    return req_types, " "
+
+@callback(
+    Output("req_type_pie_chart", "figure"),
+    Output("time_close_histogram", "figure"),
+    Output("num_req_line_chart", "figure"),
+    Output("data_quality_output", "children"),
+    Input("nc_dropdown", "value"),
+    [Input("nc_dropdown_filter", "value")],
+    Input("data_quality_switch", "value")
+)
+def generate_nc_summary_charts(nc_dropdown, nc_dropdown_filter=None, data_quality_switch=True):
+    """Generates the summary visualizations for LA 311 requests data based on selected neighborhood conucil, request types, and data quality switch.
+
+    This function takes the selected neighborhood council (nc) value from the "nc_dropdown" dropdown, selected request type from "nc_dropdown_filter"
+    dropdown, and the status on the "data_quality_switch" toggle to output a 4 main visualizations to provide an overview.
+
+    Args:
+        nc_dropdown: A string argument automatically detected by Dash callback function when "nc_dropdown" element is selected in the layout.
+        nc_dropdown_filter: A list of strings automatically detected by Dash callback function when "nc_dropdown_filter" element is selected in the layout, default None.
+        data_quality_switch: A boolean for data quality filter automatically detected by Dash callback function when "data_quality_switch" element is selected in the layout, default True.
+
+    Returns: 
+        req_type_pie_chart: pie chart that shows the share of request types out of all requests.
```

Comment:
This function seems to be returning several things that are not necessarily related. Can we have one callback per graph?

Reply:
thanks for the suggestion - resolved.